### ISSUE TITLE
Add optional randomization to points texture for particle spawn

### ISSUE THREAD

**Issue by QbieShay** [godotengine/godot-proposals#5849]
### Describe the project you are working on

Various VFX

### Describe the problem or limitation you are having in your project

I want to make particles spawn on a circle in clockwise order, but right now the particles are fetching the points at random.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Make randomization optional, specifically this line of code:
```

		code += "	int point = min(emission_texture_point_count - 1, int(rand_from_seed(alt_seed) * float(emission_texture_point_count)));\n";
```
should have the random part behind a bool in the particles material.

So when selecting emission points, the particle material will have an additional bool flag for randomization (on by default)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

see above

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It can be done with custom particle materials, but it makes it incompatible with CPU particles.

### Is there a reason why this should be core and not an add-on in the asset library?

Can't change CPUParticles.

**Comment by Calinou** [comment]
Should there also be an option to use reverse emission order when randomization is disabled? In this case, it makes sense to use an enum property with the values Random, Sequential (Normal), Sequential (Reverse).

**Comment by QbieShay** [comment]
Good catch @Calinou ! I would have done it in the texture directly but it's a very good point :+1:

**Comment by chrisb123** [comment]
Do you mean the edge of a circle, or the surface area in the circle?
The lines in emission shape sphere are what you need to change

**Comment by Calinou** [comment]
> Do you mean the edge of a circle, or the surface area in the circle?

I'm fairly sure QbieShay is referring to the edge here. In English, a "circle" refers to the edge of the circle, while a "disc" refers to the surface area.

**Comment by chrisb123** [comment]
A new emission type will also be needed, I can fix this

**Comment by chrisb123** [comment]
An additional emission type is needed. I can do this for cpu particles
